### PR TITLE
First draft of take over of existing Virtual IP 

### PR DESCRIPTION
**Pull request by wpjunior**

### CODE REVIEW

File: controller/controller.go [hunk 1]
```diff
@@ -268,10 +270,16 @@ func (r *reconcileIngress) reconcileIngress(ctx context.Context, ing *networking
 
 func (r *reconcileIngress) cleanUp(ctx context.Context, ingName types.NamespacedName, ing *networkingv1.Ingress) (reconcile.Result, error) {
 	var result reconcile.Result
-	err := r.cleanupNetworkAPI(ctx, ingName)
-	if err != nil {
-		return result, err
+
+	if takeOverVIPName := ing.Annotations[config.TakeOverAnnotation]; takeOverVIPName != "" {
+		// We wont remove the VIP, cause we use take over
```
Comment (cezarsa) [inline controller/controller.go:275]: This is where we could rollback the vip-request to its former pools as we've discussed previously, right?

File: controller/napi.go [hunk 2]
```diff
@@ -332,11 +351,53 @@ func (r *reconcileIngress) reconcileNetworkAPI(ctx context.Context, ing *network
...
 	return r.client.Status().Update(ctx, ing)
 }
+
+func (r *reconcileIngress) reconcileNetworkAPITakeOver(ctx context.Context, takeOverVIPName string, ing *networkingv1.Ingress, httpPool, httpsPool *networkapi.Pool) error {
+	lg := log.FromContext(ctx)
+
+	netapiCli := r.networkAPIClient
+	if netapiCli == nil {
+		netapiCli = networkapi.Client(r.cfg.NetworkAPIURL, r.cfg.NetworkAPIUsername, r.cfg.NetworkAPIPassword)
+	}
```
Comment (cezarsa) [inline controller/napi.go]:
```suggestion
	netapiCli := r.getNetworkAPI()
```

File: controller/napi.go [hunk 3]
```diff
@@ -332,11 +351,53 @@ func (r *reconcileIngress) reconcileNetworkAPI(ctx context.Context, ing *network
...
+	if err != nil {
+		return err
+	}
+
+	if vip.IPv4 == nil {
+		return errors.New("no ipv4 found")
+	}
+
+	vipIP, err := netapiCli.GetIPByID(ctx, vip.IPv4.ID)
+	if err != nil && !networkapi.IsNotFound(err) {
```
Comment (cezarsa) [inline controller/napi.go]: The `networkapi.IsNotFound` call along with the `vip name not found` error message should move up to after the `GetVIP` call.

File: controller/controller.go (hunk 1 above)
Comment (wpjunior) [reply controller/controller.go:275]: I figured out that the nginx-operator will replace this annotations in each reconcile loop, them, I prefer to do this manual guided by documentation.

**Pull request merged**